src/ng/interpolate.js

```diff
@@ -283,29 +284,40 @@ function $InterpolateProvider() {
         }
       }
 
+      singleExpression = concat.length === 1 && expressionPositions.length === 1;
+      parseFns = contextAllowsConcatenation && singleExpression ?
```

Comment:
Would this be any shorter/nicer doing...
```
parseFns = expressions.map(function(exp) {
    return $parse(exp, contextAllowsConcatenation && singleExpression ? undefined : parseStringifyInterceptor);
});
```

Reply:
I thought about that. It does mean evaluating that `contextAllowsConcatenation && singleExpression` for each item in `concat`. Also I was being cautious in case the `$parse` was doing parameter counting rather than just a check for `undefined`. If you feel strongly about it we could change it.

Reply:
I think it reads a little nicer but don't feel strongly about it.

Could do `var interceptor = ...` outside the loop. That might also be a little more readable which I think is most important here (I don't think performance is a major concern since this isn't run per digest or anything like that).

Reply:
I tried putting the interceptor outside the loop but it fails because the `interceptor` variable somehow gets shared between closures and breaks the tests. I followed your original suggestion.

Reply:
Done